MODULE.bazel

```diff
@@ -66,6 +66,12 @@ pip.parse(
 )
 use_repo(pip, "rules_python_publish_deps")
 
+uv = use_extension("//python:extensions.bzl", "uv")
```

Comment:
I thought it might be good to slowly evolve the repo towards a more idiomatic structure where possible. Newer rules all seem to follow a similar shape and I think it could be good for future maintainer effort if we do similar. 

The idiomatic structure can be found here:
https://github.com/bazel-contrib/rules-template/tree/main/mylang

Reply:
The template should be updated, see https://bazel.build/external/extension#put_each_extension_in_a_separate_file

The tuple `(//python:extensions.bzl, uv)` is the identity of the object. Even if you put `uv = module_extension(..)` in a common place, and re-export it in a public place (e.g. as `uv1` and `uv2` in `//python:extensions.bzl`), they're seen as distinct module extensions (e.g. they'll have separate graphs of tag_class calls).

The main issue is an extension can't see repos created by other extensions. This is true if they are defined in the same bzl file. If you end up needing one extension to load code generated by another extension (as we had to do with pip), then they _must_ go into different extension files. The final part is: because the identity is the use_extension() strings, trying to move it out of the file becomes a breaking change. But not a typical breaking change -- it affects the phase that evaluates MODULE files, so you have very limited facilities to provide a graceful upgrade path.

Reply:
Good catch! Thanks for that detailed explanation. That would have been a major disruption and a pain to fix later. Will FIXUP.